lib/ansible/module_utils/facts/system/ssh_pub_keys.py

```diff
@@ -47,6 +47,10 @@ def collect(self, module=None, collected_facts=None):
                 key_filename = '%s/ssh_host_%s_key.pub' % (keydir, type_)
                 keydata = get_file_content(key_filename)
                 if keydata is not None:
-                    ssh_pub_key_facts[factname] = keydata.split()[1]
+                    (subtype, key) = keydata.split()
+                    ssh_pub_key_facts[factname] = key
+                    if type_ == 'ecdsa':
```

Comment:
Limiting this to ecdsa means the next key type that has different sub types will require code changes in Ansible. I could see going the other way: blacklist the current ones that we know don't need the subtype. If there are others later they can be blacklisted to save space in hostvars, assuming that's the reason to limit it.